float32 x_pos
float32 y_pos
float32 x_speed
float32 y_speed
float32 ang_pos
time stamp